### PR TITLE
Use beta class to auto-hide beta oauth2 content (custom scopes)

### PR DESCRIPTION
**Pull request by royalchan-okta**
This allows us toggle it on with a beta=true query param

OKTA-93085

screen shot 2016-06-28 at 12 05 38 am

image

### CODE REVIEW

File: _source/assets/js/docs.js [hunk 1]
```diff
@@ -157,6 +157,14 @@ $(function() {
 	$(document).ready(function() {
 		$('#gen-toc-container .nav').toggleClass('hide');
 		$('#toc_current_doc').toggleClass('clicked');
+		// Unhide beta if necessary
+		var query = window.location.search.substring(1);
+        var params = query.split("&");
```
Comment (larsjohansen-okta) [inline _source/assets/js/docs.js]: nit: indentation

File: _source/docs/api/resources/oauth2.md [hunk 2]
```diff
@@ -250,6 +250,26 @@ Content-Type: application/json;charset=UTF-8
 }
 ~~~
 
+
```
Comment (larsjohansen-okta) [inline _source/docs/api/resources/oauth2.md]: will this extra line do anything?

Comment (royalchan-okta) [inline _source/docs/api/resources/oauth2.md]: yes very important

**Comment by royalchan-okta** [comment]
I've also incorporated the client credential beta changes added by @yuliu-okta , as well as my password creds and scopes cleanups from: https://github.com/okta/okta.github.io/pull/291

**Comment by cwu-okta** [comment]
@joelfranusic-okta  FYI

**Comment by larsjohansen-okta** [comment]
Shouldn't we gitignore Gemfile.lock?

File: _source/docs/api/resources/oauth-clients.md [hunk 3]
```diff
@@ -77,16 +77,18 @@ Client applications have the following properties:
...
   the OAuth 2.0 authorization code grant.
 
-> At least one redirect URI and response type is required for all client types, with one exception: if the client uses the
-  [Resource Owner Password](https://tools.ietf.org/html/rfc6749#section-4.3) flow (if `grant_types` contains the value `password`) 
-  then no redirect URI or response type is necessary. In this case you can pass either null or an empty array for these attributes.
-
+> At least one redirect URI and response type is required for all client types, with exceptions: if the client uses the
+  [Resource Owner Password](https://tools.ietf.org/html/rfc6749#section-4.3) flow (if `grant_types` contains the value `password`)        [Resource Owner Password](https://tools.ietf.org/html/rfc6749#section-4.3) flow (if `grant_types` contains the value `password`) 
+  then no redirect URI or response type is necessary. In this case you can pass either null or an empty array for these attributes.    +  or [Client Credentials](https://tools.ietf.org/html/rfc6749#section-4.4) flow (if `grant_types` contains the value `client_credentials`)
```
Comment (larsjohansen-okta) [inline _source/docs/api/resources/oauth-clients.md]: Something funky has happened here, it appears that a '+' has become part of the text.

**Comment by royalchan-okta** [comment]
Technically the Gemfile.lock is suppose to keep consistency across multiple machines as a snapshot of working module versions. It likely doesn't matter for us.

**Comment by mystiberry-okta** [comment]
Oh, I thought it was a local-only build artifact that I wasn’t supposed to check in. Now I won’t worry about it!

Thanks for the info!

Mysti

On Jun 28, 2016, at 9:03 PM, royalchan-okta <notifications@github.com<mailto:notifications@github.com>> wrote:


Technically the Gemfile.lock is suppose to keep consistency across multiple machines as a snapshot of working module versions. It likely doesn't matter for us.

**Comment by royalchan-okta** [comment]
https://github.com/okta/okta.github.io/pull/299

**Pull request closed without merging**